libraries/redshop/helper/product.php

```diff
@@ -5133,5 +5136,39 @@ public static function checkCompareProduct($productId)
         return '';
     }
 
+    /**
+     * @param   integer    $productId     Product ID
+     * @param   float    $productPrice  Product price
+     *
+     * @return  float
+     *
+     */
+    public function calculateProductApplyMassDiscount($productId, $productPrice)
+    {
+        $productId = (int)$productId;
+
+        if (!$productId) {
+            return '';
+        }
+        $product = Redshop\Product\Product::getProductById($productId);
+
+        $db    = JFactory::getDbo();
+        $query = $db->getQuery(true)
+            ->select('*')
+            ->from($db->qn('#__redshop_mass_discount'))
+            ->where(' FIND_IN_SET(' . (int)$productId . ',' . $db->quoteName('discount_product') . ')');
+
+        // Set the query and load the result.
+        $massDiscountProduct =  $db->setQuery($query)->loadObject();
+
+        //Check condition product on sale
+        if($product->product_on_sale) {
+            $productPrice = $massDiscountProduct->type == 1 ?
+                $productPrice - ($productPrice * $massDiscountProduct->amount / 100) :
+                $productPrice - $massDiscountProduct->amount;
+        }
```

Comment:
@NguyenBao10 is it currently when set masdiscount for product or category or manufacturer, the product will be check `product_on_sale`? Need you check this logic carefully